util/encoding/encoding.go

```diff
@@ -599,3 +600,57 @@ func DecodeIfNull(b []byte) ([]byte, bool) {
 	}
 	return b, false
 }
+
+// EncodeTime encodes a time value, appends it to the supplied buffer,
+// and returns the final buffer. The encoding is guaranteed to be ordered
+// Such that if t1.Before(t2) then after EncodeTime(b1, t1), and
+// EncodeTime(b2, t1), Compare(b1, b2) < 0. The time zone offset is added
+// as a two byte encoding at the end of the time encoding.
+func EncodeTime(b []byte, t time.Time) []byte {
```

Comment:
Yeah, we could check the version byte, but that seems to be poking around the internals of `Time.MarshalBinary` too much. I'd rather copy the implementation than do that. I'm not terribly concerned about the space savings of varint vs fixed-width here, though I'd note that most times can be represented by a 32-bit number of seconds (5-bytes using our varint encoding) vs the 8-bytes for storing a 64-bit seconds value as fixed-width. Similarly, the nanoseconds component will encode to a single byte in the common case of it being 0.